src/vecgf2.c

```diff
@@ -178,235 +178,22 @@ Obj AddCoeffsGF2VecGF2Vec (
-    case 26:     dothework(sptr, dptr, 26, dend); break;
-    case 27:     dothework(sptr, dptr, 27, dend); break;
-    case 28:     dothework(sptr, dptr, 28, dend); break;
-    case 29:     dothework(sptr, dptr, 29, dend); break;
-    case 30:     dothework(sptr, dptr, 30, dend); break;
-    case 31:     dothework(sptr, dptr, 31, dend); break;
-#ifdef SYS_IS_64_BIT
-    case 32:     dothework(sptr, dptr, 32, dend); break;
-    case 33:     dothework(sptr, dptr, 33, dend); break;
-    case 34:     dothework(sptr, dptr, 34, dend); break;
-    case 35:     dothework(sptr, dptr, 35, dend); break;
-    case 36:     dothework(sptr, dptr, 36, dend); break;
-    case 37:     dothework(sptr, dptr, 37, dend); break;
-    case 38:     dothework(sptr, dptr, 38, dend); break;
-    case 39:     dothework(sptr, dptr, 39, dend); break;
-    case 40:     dothework(sptr, dptr, 40, dend); break;
-    case 41:     dothework(sptr, dptr, 41, dend); break;
-    case 42:     dothework(sptr, dptr, 42, dend); break;
-    case 43:     dothework(sptr, dptr, 43, dend); break;
-    case 44:     dothework(sptr, dptr, 44, dend); break;
-    case 45:     dothework(sptr, dptr, 45, dend); break;
-    case 46:     dothework(sptr, dptr, 46, dend); break;
-    case 47:     dothework(sptr, dptr, 47, dend); break;
-    case 48:     dothework(sptr, dptr, 48, dend); break;
-    case 49:     dothework(sptr, dptr, 49, dend); break;
-    case 50:     dothework(sptr, dptr, 50, dend); break;
-    case 51:     dothework(sptr, dptr, 51, dend); break;
-    case 52:     dothework(sptr, dptr, 52, dend); break;
-    case 53:     dothework(sptr, dptr, 53, dend); break;
-    case 54:     dothework(sptr, dptr, 54, dend); break;
-    case 55:     dothework(sptr, dptr, 55, dend); break;
-    case 56:     dothework(sptr, dptr, 56, dend); break;
-    case 57:     dothework(sptr, dptr, 57, dend); break;
-    case 58:     dothework(sptr, dptr, 58, dend); break;
-    case 59:     dothework(sptr, dptr, 59, dend); break;
-    case 60:     dothework(sptr, dptr, 60, dend); break;
-    case 61:     dothework(sptr, dptr, 61, dend); break;
-    case 62:     dothework(sptr, dptr, 62, dend); break;
-    case 63:     dothework(sptr, dptr, 63, dend); break;
-#endif
-    default:  Pr("Illegal shift %i", cbits, 0);
-      SyExit(2);
-    }
+  soff = (smin-1) %BIPEB;
+  doff = (dmin-1) %BIPEB;
+  sptr = BLOCKS_GF2VEC(src) + (smin-1)/BIPEB;
+  dptr = BLOCKS_GF2VEC(dest) + (dmin-1)/BIPEB;
 
-    /* fixup pointers */
-    sptr += (dend - dptr);
-    dptr = dend;
-    /* OK, so now we may need to copy some more bits to fill the final block */
-    endbits = (dmin + nelts) % BIPEB;
-    if (endbits) 
-      {
-	if (endbits <= cbits)
-	  setlowbits(dptr, endbits, midbits(*sptr++,BIPEB-cbits, endbits));
-	else {
-	  bits = highbits(*sptr++,cbits);
-	  setlowbits(dptr, endbits, bits | (lowbits(*sptr, endbits - cbits) << cbits));
-	}
-      }
-    return;
-  }
+  CopyBits(sptr, soff, dptr, doff, nelts);
```

Comment:
Have you conducted tests as to whether (and if so: how exactly) this change to the GF2 code affects performance?